app/src/main/kotlin/pl/gov/mc/protego/ui/registration/RegistrationActivity.kt

```diff
@@ -23,29 +27,36 @@ import pl.gov.mc.protego.ui.validator.MsisdnOk
 
 class RegistrationActivity : BaseActivity() {
 
-    private val registrationViewModel: RegistrationViewModel by viewModel()
+    override val viewModel: RegistrationViewModel by viewModel()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.registration_view)
 
         register_button.isEnabled = false
         register_button.setOnClickListener {
-            registrationViewModel.onStartRegistration(msisdn_edit_text.text.toString().replace(" ", ""))
+            viewModel.onStartRegistration(msisdn_edit_text.text.toString().replace(" ", ""))
         }
 
-        msisdn_edit_text.onTextChanged(registrationViewModel::onNewMsisdn)
+        skip_registration_button.setOnClickListener {
+            viewModel.onSkipRegistrationClicked()
+        }
+
+        msisdn_edit_text.onTextChanged(viewModel::onNewMsisdn)
         msisdn_edit_text.scrollWhenFocusObtained(scroll_view)
 
         supportActionBar?.apply {
             setHomeButtonEnabled(true)
             setDisplayHomeAsUpEnabled(true)
         }
 
+        setupLinkToTermsOfUse()
         observeMsisdnValidation()
         observeRegistrationStatus()
+        observeIntents()
+        observeIsInProgress()
```

Comment:
moim zdaniem, lepsza była by bardziej opisowa nazwa. Co jest w `inProgress`?

Reply:
Mechanizm jest generyczny po stronie viewmodelu, dlatego tam się nazywa `isInProgress` - może to request sieciowy, może to pobieranie z bazy danych, może odpalanie BT, chyba nie robi różnicy. Zmiana nazwy w aktywności spowoduje oderwanie od nazwy w `BaseViewModel`. Może `isInBlockingProgress` ewentualnie?

Reply:
Do poprawienia na parametryzowany typ w osobnym PR.

Reply:
Wyniesione do #93